### PR TITLE
doc: Add documentation for AVSystem integration

### PR DESCRIPTION
**Pull request by umapraseeda**
Add documentation for AVSystem integration
NCSDK-17738

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

### CODE REVIEW

**Review by SeppoTakalo — CHANGES_REQUESTED** [review]
Looks good.

I left a few notes that I propose to be changed. Otherwise I think the content is fine.

File: doc/nrf/ug_avsystem.rst [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+The integration supports the following location services:
+
+* Cell-based location assistance
+* A-GPS location assistance
+
+Cell-based location
+===================
+
+You can use a supported sample to determine the cell-based location of the device if you have nRF Cloud integration with AVSystem.
+Cell-based location offers accuracy up to 1000 meters.
```
Comment (VeijoPesonen) [inline doc/nrf/ug_avsystem.rst:126]: In the case of single- or multicell? I assume that at least with multicell the accuracy is better than a kilometer.

File: doc/nrf/ug_avsystem.rst [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+After you flash the supported sample or application on to your device, the device requests A-GPS data.
+`Coiote Device Management server`_ obtains the data from nRF Cloud and sends the A-GPS data blob to the GNSS module of the device.
+This data is used in combination with the GPS data to acquire a fix.
+
+.. figure:: images/avsystem_integration_agps_location.svg
+   :alt: AVSystem integration - A-GPS location assistance
+
+   AVSystem integration - A-GPS location assistance
+
+For cell-based location assistance, each supported sample might use a different overlay file.
```
[the remark this answers is not in the document]
Comment (umapraseeda) [reply doc/nrf/ug_avsystem.rst:159]: Fixing this for the cell-based image also.

File: doc/nrf/ug_avsystem.rst (hunk 1 above)
Comment (umapraseeda) [reply doc/nrf/ug_avsystem.rst]: Removed this as it is already explained in https://docs.nrfcloud.com/LocationServices/LocationServicesOverview/#types.

File: doc/nrf/ug_avsystem.rst [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      Refer to the **Configuration files** section in the respective sample or application documentation for more information.
+
+After flashing the sample, reboot your device for reconnection.
+From the terminal, you can see that the registration is complete.
+You can also check the status on the Coiote Device Management UI.
+On the `Coiote Device Management UI <Coiote Device Management server_>`_, after adding your device, you can see a ``Device connected successfully`` message if the device was added successfully.
+Click :guilabel:`Next`, :guilabel:`Finish` and :guilabel:`Go to device`.
+You can see that the device is shown as registered under :guilabel:`Registration status` in **Device Center**.
+
+nRF Cloud integration with AVSystem
```
Comment (divipillai) [inline doc/nrf/ug_avsystem.rst:112]: Should this section be part of the numbered list?

File: samples/nrf9160/lwm2m_client/sample_description.rst [hunk 4]
```diff
@@ -283,6 +272,30 @@ Following are the instructions for enabling notifications in the Leshan Demo ser
...
+  #. Set :kconfig:option:`CONFIG_APP_LWM2M_PSK` to an empty string.
+
+* Using |VSC|:
+
+  a. In |VSC|, select `Add an existing application <Migrating IDE_>`_ and select the sample folder.
+  #. Under **Actions**, click :guilabel:`Kconfig`.
+  #. Click :guilabel:`Application sample`.
+  #. Under **LwM2M objects**, remove the key value next to :guilabel:`LwM2M pre-shared key for communication`.
+  #. Save and close the configuration.
+
```
Comment (divipillai) [inline samples/nrf9160/lwm2m_client/sample_description.rst:293]: We could make these as tabs (.. tabs::)

**Review by divipillai — COMMENTED** [review]
Looks good to me. Some minor suggestions

**Review by coderbyheart — CHANGES_REQUESTED** [review]

**Review by coderbyheart — APPROVED** [review]

File: doc/nrf/ug_avsystem.rst [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+* :ref:`asset_tracker_v2` - The :ref:`asset_tracker_v2_cloud_module` is set to communicate with AVSystem's Coiote Device Management, with a runtime provisioned Pre-shared key (PSK) set by the ``CONFIG_LWM2M_INTEGRATION_PSK`` Kconfig option.
+  For more information, see :ref:`assettrcker_v2_cloudmodule_lwm2m`.
+
+The following samples uses the AVSystem integration in |NCS|:
+
+* :ref:`lwm2m_client`
+* :ref:`lwm2m_carrier`
+* :ref:`modem_shell_application`
+
```
[the remark this answers is not in the document]
Comment (umapraseeda) [reply doc/nrf/ug_avsystem.rst:184]: Added

**Review by VeijoPesonen — APPROVED** [review]

**Review by tokangas — COMMENTED** [review]
Looks good, just a couple of comments about usage of term GPS vs. GNSS.

`avsystem_integration_agps_location.svg` should also be updated, `GPS` -> `GNSS`.

**Review by tokangas — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

File: doc/nrf/ug_avsystem.rst [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+All these samples offer LwM2M support and can be used to fetch location assistance data from nRF Cloud by proxying it through AVSystem's Coiote LwM2M server.
+
+Library support
+***************
+
```
Comment (richabp) [inline doc/nrf/ug_avsystem.rst:189]:
```suggestion

The following |NCS| library supports the AVSystem integration:

```

File: doc/nrf/ug_avsystem.rst [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+Cell-based location
+===================
+
+You can use a supported sample to determine the cell-based location of the device if you have nRF Cloud integration with AVSystem.
+Both single cell and multi cell location services can be used.
+See :ref:`location_assistance_cell` for more information on the API usage.
+
+After you flash the sample, the device starts searching for neighboring cell towers.
+After the completion of registration in Coiote Device Management server is complete, the sample passes the information onto `Coiote Device Management server`_ which then forwards it as a cell-location query to nRF Cloud.
```
Comment (richabp) [inline doc/nrf/ug_avsystem.rst:134]:
```suggestion
After the completion of registration in Coiote Device Management server, the sample passes the information onto `Coiote Device Management server`_ which then forwards it as a cell-location query to nRF Cloud.
```

File: doc/nrf/ug_avsystem.rst [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+AVSystem integration
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+AVSystem is a software solution provider that provides a device management platform.
+This platform is based on the LwM2M protocol with |NCS| compatibility for Nordic Semiconductor's cellular IoT products.
+It handles the entire process from onboarding till decommissioning of devices and its integration with |NCS| ensures secure and power efficient device management solutions using Nordic Semiconductor's cellular devices.
```
Comment (richabp) [inline doc/nrf/ug_avsystem.rst:12]: Should we split it into two sentences and add a blank line to improve the readability?
```suggestion
It handles the entire process from onboarding till decommissioning of devices.

Its integration with |NCS| ensures secure and power efficient device management solutions using Nordic Semiconductor's cellular devices.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by divipillai — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Comment by VeijoPesonen** [comment]
@mierunski Please have a look if there's anything you want to add or modify.

**Pull request merged**